gapic-common/lib/gapic/rest/paged_enumerable.rb

```diff
@@ -0,0 +1,225 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Gapic
+  module Rest
+    ##
+    # A class to provide the Enumerable interface to the response of a REST paginated method.
+    # PagedEnumerable assumes response message holds a list of resources and the token to the next page.
+    #
+    # PagedEnumerable provides the enumerations over the resource data, and also provides the enumerations over the
+    # pages themselves.
+    #
+    # @example normal iteration over resources.
+    #   paged_enumerable.each { |resource| puts resource }
+    #
+    # @example per-page iteration.
+    #   paged_enumerable.each_page { |page| puts page }
+    #
+    # @example Enumerable over pages.
+    #   paged_enumerable.each_page do |page|
+    #     page.each { |resource| puts resource }
+    #   end
+    #
+    # @example more exact operations over pages.
+    #   while some_condition()
+    #     page = paged_enumerable.page
+    #     do_something(page)
+    #     break if paged_enumerable.next_page?
+    #     paged_enumerable.next_page
+    #   end
+    #
+    class PagedEnumerable
+      include Enumerable
+
+      ##
+      # @attribute [r] page
+      #   @return [Page] The current page object.
```

Comment:
An `@attribute` tag is normally used on the class rather than on the attribute itself.

Reply:
:+1: